CMakeLists.txt

```diff
@@ -12,29 +12,31 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -pedantic -Wold-style-cast -I ${CMAKE_CURRENT_SOURCE_DIR}/Lib/")
 endif()
 
-set(MPL_GROUP
-    Lib/Mpl/Types.hpp
-    Lib/Mpl/Algorithm.hpp
-    Lib/Mpl/IntegralConstants.hpp
-    Lib/Mpl/Utility.hpp
-)
-
-set(REGISTER_GROUP
-    Lib/Register/Apply.hpp
-    Lib/Register/AtomicFactories.hpp
-    Lib/Register/Exec.hpp
-    Lib/Register/Factories.hpp
-    Lib/Register/IsolatedFactories.hpp
-    Lib/Register/Register.hpp
-    Lib/Register/Seam.hpp
-    Lib/Register/Types.hpp
-    Lib/Register/Utility.hpp
-)
-
-add_library(Kvasir ${MPL_GROUP} ${REGISTER_GROUP})
-target_include_directories(Kvasir PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/Lib)
-set_target_properties(Kvasir PROPERTIES LINKER_LANGUAGE CXX)
-target_compile_definitions(Kvasir PUBLIC "DBRIGAND_NO_BOOST_SUPPORT")
+#set(MPL_GROUP
+#    Lib/Mpl/Types.hpp
+#    Lib/Mpl/Algorithm.hpp
+#    Lib/Mpl/IntegralConstants.hpp
+#    Lib/Mpl/Utility.hpp
+#)
+#
+#set(REGISTER_GROUP
+#    Lib/Register/Apply.hpp
+#    Lib/Register/AtomicFactories.hpp
+#    Lib/Register/Exec.hpp
+#    Lib/Register/Factories.hpp
+#    Lib/Register/IsolatedFactories.hpp
+#    Lib/Register/Register.hpp
+#    Lib/Register/Seam.hpp
+#    Lib/Register/Types.hpp
+#    Lib/Register/Utility.hpp
+#)
+#
+#add_library(Kvasir ${MPL_GROUP} ${REGISTER_GROUP})
+
+add_library(Kvasir INTERFACE)
+target_include_directories(Kvasir INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/Lib)
+#set_target_properties(Kvasir PROPERTIES LINKER_LANGUAGE CXX)
+target_compile_definitions(Kvasir INTERFACE "DBRIGAND_NO_BOOST_SUPPORT")
```

Comment:
This doesn't seem right, shouldn't the line be

    target_compile_definitions(Kvasir INTERFACE BRIGAND_NO_BOOST_SUPPORT)

Or does the gating symbol really begin with "D"?